### DISCUSSION TITLE
Find not detecting field changes by third party

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#2314] [Q&A] [answered]
I have a repository and I'm trying to use a findAndCount on it.
I'm ommiting fields attribute to get all fields and it works fine until I make any update on a record without using Mikro-ORM.
If I change a field that is not being used on the filter straight at the database (or using a third party), it does not detect it.

Although if I change the filtered field or update it using the same repository update method, it detects the change.

For example. I have this "SpotRadio" repository with a findAndCount method:

```
import ISpotRadioRepository from '../../../repositories/ISpotRadioRepository';
import IRadiosRepository from '../../../repositories/IRadiosRepository';
import { EntityRepository } from '@mikro-orm/mysql';
import SpotRadio from '../entities/SpotRadio';
import { inject, injectable } from 'tsyringe';
import IDatabaseProvider from '@shared/container/providers/DatabaseProvider/models/IDatabaseProvider';
import AppError from '@shared/errors/AppError';

@injectable()
class SpotRadioRepository implements ISpotRadioRepository {
  private ormRepository: EntityRepository<SpotRadio>;

  constructor(
    @inject('DatabaseProvider')
    private databaseProvider: IDatabaseProvider,
    @inject('RadiosRepository')
    private radiosRepository: IRadiosRepository,
  ) {
    this.ormRepository = this.databaseProvider
      .getConnection()
      .em.getRepository(SpotRadio);
  }

  public async findAll(
    status: string,
    page: number,
  ): Promise<[SpotRadio[], number]> {
    const spots = await this.ormRepository.findAndCount(
      { status, deleted_at: null },
      {
        orderBy: { id: 'desc' },
        populate: ['editor', 'radios'],
        limit: 20,
        offset: 20 * (page - 1),
      },
    );

    return [
      spots[0],
      1 + spots[1] / 20,
    ];
  }
```

If I change any field **directly on database**, when I call that route, it is **not detecting the change**.
But if I update that same field using **mikro-orm update**, it **detects the change**.
And if I change the deleted_at or **status field (which are being used on filter) directly on database**, it **detects the change**.
Also, if I restart the whole back-end, it detects the change

I tried to specify every field using fields attribute and I tried to disable caching, but it didn't change anything, 
Is this an issue? Am I missing something?

**Comment by B4nan** [comment, marked as answer]
Do you have request context set up? Each request handler needs to have its own context (own identity map).

**Reply by kennethsato** [reply]
So sorry @B4nan !
I wasn't passing the requestContext middleware for this route.
I've added it and it's working!
Thank you!

But something else misled me to not noticing the missing requestContext.
As it is not related to this title, I think I'll create another discussion for that if you do not mind.
